src/pages/index.js

```diff
@@ -0,0 +1,349 @@
+);
+const avatarModal = document.querySelector("#avatar-modal");
+
+// avatar form elements
+const avatarSubmitButton = avatarModal.querySelector(".modal__submit-button");
+const avatarCardElement = avatarModal.querySelector(".modal__form");
+const addAvatarModalCloseButton = avatarModal.querySelector(
+  ".modal__close-button"
+);
+const avatarLinkInput = avatarModal.querySelector("#profile-avatar-input");
+
+api.getAppInfo()
+.then(([cards, userInfo]) => {
+  console.log("Fetched cards:", cards);
+  console.log("Number of cards:", cards.length);
+  cards.forEach((item) => {
+    const cardElement = getCardElement(item);
+    cardsList.prepend(cardElement);
+  });
+  profileNameElement.textContent = userInfo.name;
+  profileDescriptionElement.textContent = userInfo.about;
+  avatar.src = userInfo.avatar;
+})
+.catch(console.error);
+
+// function to handle closing the modal with escape key
+function handleEscapeKey(event) {
+  if (event.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) {
+      closeModal(openedModal);
+    }
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown",handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown",handleEscapeKey);
+}
+
+
+//add listeners for closing the modal on overlay click
+function addOverlayClickListener(modal) {
+  modal.addEventListener("click", (event) => {
+    if (event.target === modal) {
+      closeModal(modal);
+    }
+  });
+}
+
+// select all modals
+const modals = document.querySelectorAll(".modal");
+
+avatarModalButton.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+avatarCardElement.addEventListener("submit", handleAvatarSubmit);
+
+// add listener to each modal
+modals.forEach((modal) => {
+  addOverlayClickListener(modal);
+})
+
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value
+    })
+    .then((data) => {
+      profileNameElement.textContent = data.name;
+      profileDescriptionElement.textContent = data.about;
+      avatar.src = data.avatar;
+
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+
+  api
+    .addCard({
+      name: inputValues.name,
+      link: inputValues.link
+    })
+    .then((newCard) => {
+      console.log("New card added:", newCard);
+      const cardElement = getCardElement(newCard);
+      cardsList.prepend(cardElement);
```

Comment:
**Could be improved**

You can make a universal function for adding a card into the section using different methods: `append`, `prepend`, or anything else:

```js

// The function accepts a card object and a method of adding to the section
// The method is initially `prepend`, but you can pass `append` 
function renderCard(item, method = "prepend") {
  
  const cardElement = getCardElement(item);
  // Add the card into the section using the method
  cardsList[ method ](cardElement);
}



```